test/image/mocks/bar_attrs_relative.json

```diff
@@ -2,27 +2,37 @@
     "data":[
         {
             "width":[1,0.8,0.6,0.4],
+            "text":[1,2,3333333333,4],
+            "textposition":"outside",
             "y":[1,2,3,4],
             "x":[1,2,3,4],
             "type":"bar"
         }, {
             "width":[0.4,0.6,0.8,1],
+            "text":["Three",2,"outside text",0],
```

Comment:
> Why is "outside text" outside? Wouldn't it be just as big on the inside?

The way I understood the criteria for `auto` is that if it doesn't fit inside, it goes outside (no matter if shrinking is required to draw it outside).

> Where did "0" go?

0-size bars aren't even included in `gd.calcdata`

> it feels like only trace 1 should be able to have a label there

I think this feeling is specific to this case, because `trace[1]` only has bars with positive sizes. If `trace[1][3]` was `-1` instead of `0`, then the bar and the label would be drawn on the negative side.